### PR TITLE
Restore Array surface area

### PR DESCRIPTION
**Pull request by sepidehMS**
Finishing up https://github.com/dotnet/corefx/pull/10256 while @danmosemsft is out. (Issue https://github.com/dotnet/corefx/issues/9998)

ApiCompat still fails on this when checking whether facade points to implementation.

### CODE REVIEW

**Comment by sepidehMS** [comment]
@weshaggard @ericstj @stephentoub Any idea on what I've done wrong that these 2 builds failed?

ApiCompat fails with "member does not exist in the implementation but does exist in the contract". Could it be because newly added members were not exposed [here](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/ref/mscorlib.cs)?

**Comment by weshaggard** [comment]
@sepidehMS yes that would be the reason. Can you rebase your changes on the latest master commit? I think some of these are already added but that will give you a more definitive list of what is missing and may need to be added.

**Comment by sepidehMS** [comment]
@dotnet-bot test this please

**Comment by sepidehMS** [comment]
@dotnet-bot test this please

**Comment by sepidehMS** [comment]
@weshaggard The build cannot find the definition of newly added Array members.
These are the commits made in CoreClr for this issue: https://github.com/dotnet/coreclr/commit/3e9d46373f7067f140c8a2881d87c82d132727da, https://github.com/dotnet/coreclr/commit/0d6291d3d92fd0c4fb97a23149243b9dd322cf27
Is anything else needed to expose these members?

**Comment by sepidehMS** [comment]
@dotnet-bot test this please

**Comment by sepidehMS** [comment]
Updated based on feedback.

cc: @weshaggard @stephentoub @ericstj @danmosemsft

File: src/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -65,17 +65,19 @@ public partial class ArithmeticException : System.Exception
         public ArithmeticException(string message) { }
         public ArithmeticException(string message, System.Exception innerException) { }
     }
-    public abstract partial class Array : System.Collections.ICollection, System.Collections.IEnumerable, System.Collections.IList, System.Collections.IStructuralComparable, System.Collections.IStructuralEquatable
+    public abstract partial class Array : System.Collections.ICollection, System.Collections.IEnumerable, System.Collections.IList, System.Collections.IStructuralComparable, System.Collections.IStructuralEquatable, System.ICloneable
```
Comment (weshaggard) [inline src/System.Runtime/ref/System.Runtime.cs]: Can you please verify that this represents all of the Array members now? Are there any more that aren't included here?

**Comment by weshaggard** [comment]
Other then the one question looks good to me.

File: src/System.Runtime/ref/System.Runtime.cs (hunk 1 above)
Comment (sepidehMS) [inline src/System.Runtime/ref/System.Runtime.cs]: @weshaggard Yes, all Array members are exposed now.

Comment (sepidehMS) [inline src/System.Runtime/ref/System.Runtime.cs]: @avjgit Yes, it has been exposed from [System.Runtime](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/ref/System.Runtime.cs) contract.

**Pull request merged**